### ISSUE TITLE
Finding similarity

### ISSUE THREAD

**Issue by bsmar** [cjmay/athena#1]
Could you provide a little bit more info, how to find similarity for specific word.
I've trained model via word2vec-train on small (100MB) text corpus (http://mattmahoney.net/dc/text8.zip)

And now i want to test finding similarity.
I tried test it with https://github.com/tmikolov/word2vec
But when i run ./distance <path to model> - and typing any word, getting "Out of dictionary word!"

**Comment by ccmaymay** [comment]
Hi @bsmar,

If using the `master` version of the code, you can write out your model in the word2vec text format using `word2vec-print`, and you should be able to run `distance` on that file.  This format is one word per line (except for the first line, which shows the number of words and embedding size), so you might want to open the file in an editor (or page through it using `less`) before running `distance` on it to check that it looks okay.

Hope this helps,
Chandler

**Comment by bsmar** [comment]
Thanks for your response, it helped!

**Comment by bsmar** [comment]
Hello @cjmay 
It is not clear to me, can I use this lib to continue training already pre-trained model with additional data?
For example:
I've used spacesaving-word2vec-train script to train model on one data.
And after some time I want to continue train this model with additional data.

Thank you.

**Comment by ccmaymay** [comment]
Hi @bsmar,

I'm sorry there is not better high-level documentation.  I believe you can load a model created by `spacesaving-word2vec-train` and resume training, but you will need to write a new script to do so---there is not an existing script for that.

Note that at the end of `spacesaving-lm-train` there is a [call to FileSerializer::dump](https://github.com/cjmay/athena/blob/master/src/spacesaving-lm-train.cpp#L116).  In `spacesaving-lm-print` there is a corresponding [call to FileSerializer::load](https://github.com/cjmay/athena/blob/master/src/spacesaving-lm-print.cpp#L64).

The spacesaving word2vec model is basically some extra stuff on top of the `SpaceSavingLanguageModel` (see [this typedef in spacesaving-word2vec-train](https://github.com/cjmay/athena/blob/master/src/spacesaving-word2vec-train.cpp#L31)) and you can see that [spacesaving-word2vec-train uses the same FileSerializer interface as spacesaving-lm-train](https://github.com/cjmay/athena/blob/master/src/spacesaving-word2vec-train.cpp#L208) to save the spacesaving word2vec model to disk.  So I think you can copy [spacesaving-word2vec-train.cpp](https://github.com/cjmay/athena/blob/master/src/spacesaving-word2vec-train.cpp) to a new file (or modify it directly if you wish), replace some of the existing object initializations with `FileSerializer::load`, and be just about ready to go.

If you choose to add a new file, just add an entry to [MAIN_SOURCES in the Makefile](https://github.com/cjmay/athena/blob/master/Makefile#L80) to add it to the build.

Note that this will (I think) give you a spacesaving word2vec model with the exact same state as where you left off---so for example your learning rate will be reinitialized to the final learning rate from your [previous training run](https://github.com/cjmay/athena/blob/master/src/_core.cpp#L503).  If that's not what you want, you may need to do some extra fiddling.  It has been a while since I worked on this code; I don't have everything fresh in my memory.  Sorry I don't have time to work on this myself right now, it's a very reasonable feature to want.

Would it be okay if I closed this issue, since your question about similarity was resolved?  Please feel free to open other issues if your new issue isn't resolved, or you have other questions.

Chandler

**Comment by ccmaymay** [comment]
(I've added a new issue for resuming from a saved model: #2)

**Comment by bsmar** [comment]
Thank you for detail response!
Sure it would be okay if you close this issue.
